flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskMultipleInputSelectiveReadingTest.java

```diff
@@ -197,22 +198,21 @@ public void testInputStarvation() throws Exception {
 			testHarness.processElement(new StreamRecord<>("3"), 1);
 			testHarness.processElement(new StreamRecord<>("4"), 1);
 
-			testHarness.processSingleStep();
 			expectedOutput.add(new StreamRecord<>("[2]: 1"));
-			testHarness.processSingleStep();
 			expectedOutput.add(new StreamRecord<>("[2]: 2"));
-			assertThat(testHarness.getOutput(), contains(expectedOutput.toArray()));
+			testHarness.processAll();
+			assertEquals(expectedOutput, new ArrayList<>(testHarness.getOutput()).subList(0, expectedOutput.size()));
```

Comment:
@pnowojski, 
this test started to fail after rebasing of `Read channel state unconditionally`.
So I replaced `processSingleStep` with `processAll`.
Please take a look.

Reply:
This test is now not doing what it was intended.

Now you are processing all elements from the input gate `1` before `testHarness.processElement(new StreamRecord<>("1"), 2);` (L207/206) is being enqueued to input gate `2`.

I would guess that 
```
// to avoid starvation, if the input selection is ALL and availableInputsMask is not ALL,
// always try to check and set the availability of another input
if (inputSelectionHandler.shouldSetAvailableForAnotherInput()) {
	fullCheckAndSetAvailable();
}
```
check from `StreamMultipleInputProcessor#selectNextReadingInputIndex` is currently not tested.

The intention behind this test is:
1. to have a long (just as well could be infinite) backlog of records to process on one of the inputs
2. introduce the availability change on the second input, and make sure it's checked/respected (instead of hot looping on the first input)
3. also throw in a third not selected input just to spice things a little bit

Why did you have to change this test?

Reply:
Thanks for the explanation.
The problem was that the output didn't match the expected one. The reason was that requesting partitions now also requires a mailbox step. 
After your comment I replaced the fix with:
```
for (int i = 0; i < testHarness.inputGates.length; i++) {
	testHarness.processSingleStep();
}
```
Does it look good to you?

Reply:
Hmmm, maybe replace it with more reliable condition, sth like:
```
boolean noElementFromInputGate3 = true;
steps = 0;
while (noElementFromInputGate3 && steps++ < 100 && testHarness.procesSingleStep()) {
  noElementFromInputGate3 = ... // check for presence of `new StreamRecord<>("[3]: 1"))`
}
```
?